CMakeLists.txt

```diff
@@ -66,76 +34,113 @@ option(HIGHFIVE_HAS_WERROR "Convert warnings to errors." OFF)
 option(HIGHFIVE_HAS_FRIEND_DECLARATIONS "Enable additional friend declarations. Certain compiler require this On, others Off." OFF)
 mark_as_advanced(HIGHFIVE_HAS_FRIEND_DECLARATIONS)
 
-set(HIGHFIVE_SANITIZER OFF CACHE STRING "Enable a group of sanitizers, requires compiler support. Supported: 'address' and 'undefined'.")
-mark_as_advanced(HIGHFIVE_SANITIZER)
+# Configure Tests & Examples
+# --------------------------
+
+# Internal variable that controls the default value for building
+# optional things like tests, examples and docs.
+if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+  set(HIGHFIVE_EXTRAS_DEFAULT ON)
+else()
+  set(HIGHFIVE_EXTRAS_DEFAULT OFF)
+endif()
+
+option(HIGHFIVE_UNIT_TESTS "Compile unit-tests" ${HIGHFIVE_EXTRAS_DEFAULT})
+option(HIGHFIVE_EXAMPLES "Compile examples" ${HIGHFIVE_EXTRAS_DEFAULT})
+option(HIGHFIVE_BUILD_DOCS "Build documentation" ${HIGHFIVE_EXTRAS_DEFAULT})
+
+option(HIGHFIVE_USE_BOOST "Enable Boost testing" OFF)
```

Comment:
The fear is that CI will be green; but nothing ran because boost is spelled `BooST` or it's moved somewhere in the image and CMake can't find it anymore, or ...

Yes, name could be changed, again it's very much WIP.

Reply:
Renamed, but kept the variables for the reason given.